main/EDTypesMod.F90

```diff
@@ -547,5 +549,48 @@ function get_current_landuse_statevector(this) result(current_state_vector)
      end do
 
    end function get_current_landuse_statevector
-  
+
+   ! =====================================================================================
+
+   function get_secondary_young_fraction(this) result(secondary_young_fraction)
+
+     !
+     ! !DESCRIPTION:
+     !  Calculate how much of the secondary area is "young", i.e. below the age threshold.
+     !  If no seconday patch area at all, return -1.
+     !
+     ! !USES:
+     !
+     ! !ARGUMENTS:
+     class(ed_site_type) :: this
+     real(r8)            :: secondary_young_fraction
+     real(r8)            :: secondary_young_area
+     real(r8)            :: secondary_old_area
+
+     ! !LOCAL VARIABLES:
+     type(fates_patch_type), pointer :: currentPatch
+
+     secondary_young_area = 0._r8
+     secondary_old_area = 0._r8
```

Comment:
Is "secondary mature" better than "secondary old"? Just want to keep some naming convention to be consistent.